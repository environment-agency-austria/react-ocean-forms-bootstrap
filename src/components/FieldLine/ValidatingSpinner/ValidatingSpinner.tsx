/**
 * Copyright (c) 2018-present, Umweltbundesamt GmbH
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import * as React from 'react';

import { faSpinner } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import { IValidatingSpinnerProps } from './ValidatingSpinner.types';

/**
 * Displays an error icon if valid is false
 */
export class ValidatingSpinner extends React.Component<IValidatingSpinnerProps> {
  public static displayName: string = 'ValidatingSpinner';

  public render(): JSX.Element | null {
    const { isValidating } = this.props;

    if (!isValidating) { return null; }

    return (
      <FontAwesomeIcon
        icon={faSpinner}
        spin
        className="ico-loading mr-2"
      />
    );
  }
}
