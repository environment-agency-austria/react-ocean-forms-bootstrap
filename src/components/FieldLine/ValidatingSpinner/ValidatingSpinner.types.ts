/**
 * Props for the ValidatingSpinner component
 */
export interface IValidatingSpinnerProps {
  /**
   * True if the Field is currently validating
   * and thus the spinner should be visible
   */
  isValidating: boolean;
}
