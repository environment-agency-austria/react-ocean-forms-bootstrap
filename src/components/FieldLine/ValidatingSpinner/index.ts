export * from './ValidatingSpinner';
export * from './ValidatingSpinner.types';
