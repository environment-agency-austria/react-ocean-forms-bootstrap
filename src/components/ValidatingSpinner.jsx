/**
 * Copyright (c) 2018-present, Umweltbundesamt GmbH
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import PropTypes from 'prop-types';
import FontAwesomeIcon from '@fortawesome/react-fontawesome';
import faSpinner from '@fortawesome/fontawesome-free-solid/faSpinner';

/**
 * Displays an error icon if valid is false
 */
function ValidatingSpinner({ isValidating }) {
  return isValidating && <FontAwesomeIcon icon={faSpinner} spin className="ico-loading mr-2" />;
}

ValidatingSpinner.propTypes = {
  isValidating: PropTypes.bool,
};

export default ValidatingSpinner;
