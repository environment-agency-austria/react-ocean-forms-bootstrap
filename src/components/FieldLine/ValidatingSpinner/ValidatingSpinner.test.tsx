import * as React from 'react';

import { shallow, ShallowWrapper } from 'enzyme';

import { ValidatingSpinner } from './ValidatingSpinner';

describe('<ValidatingSpinner />', () => {
  const setup = (isValidating: boolean): ShallowWrapper => shallow((
    <ValidatingSpinner
      isValidating={isValidating}
    />
  ));

  it('should render nothing if isValidating is false', () => {
    const wrapper = setup(false);
    expect(wrapper.find('FontAwesomeIcon').exists()).toBeFalsy();
  });

  it('should render a FontAwesomeIcon if isValidating is true', () => {
    const wrapper = setup(true);
    expect(wrapper.find('FontAwesomeIcon').exists()).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });
});
