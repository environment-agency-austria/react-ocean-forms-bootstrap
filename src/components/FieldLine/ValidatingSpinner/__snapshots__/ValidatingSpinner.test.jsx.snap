// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ValidatingSpinner /> should render a FontAwesomeIcon if isValidating is true 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ValidatingSpinner
    isValidating={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "border": false,
      "className": "ico-loading mr-2",
      "fixedWidth": false,
      "flip": null,
      "icon": Object {
        "icon": Array [
          512,
          512,
          Array [],
          "f110",
          "M304 48c0 26.51-21.49 48-48 48s-48-21.49-48-48 21.49-48 48-48 48 21.49 48 48zm-48 368c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm208-208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zM96 256c0-26.51-21.49-48-48-48S0 229.49 0 256s21.49 48 48 48 48-21.49 48-48zm12.922 99.078c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.491-48-48-48zm294.156 0c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.49-48-48-48zM108.922 60.922c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.491-48-48-48z",
        ],
        "iconName": "spinner",
        "prefix": "fas",
      },
      "inverse": false,
      "listItem": false,
      "mask": null,
      "pull": null,
      "pulse": false,
      "rotation": null,
      "size": null,
      "spin": true,
      "symbol": false,
      "title": "",
      "transform": null,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "border": false,
        "className": "ico-loading mr-2",
        "fixedWidth": false,
        "flip": null,
        "icon": Object {
          "icon": Array [
            512,
            512,
            Array [],
            "f110",
            "M304 48c0 26.51-21.49 48-48 48s-48-21.49-48-48 21.49-48 48-48 48 21.49 48 48zm-48 368c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm208-208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zM96 256c0-26.51-21.49-48-48-48S0 229.49 0 256s21.49 48 48 48 48-21.49 48-48zm12.922 99.078c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.491-48-48-48zm294.156 0c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.49-48-48-48zM108.922 60.922c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.491-48-48-48z",
          ],
          "iconName": "spinner",
          "prefix": "fas",
        },
        "inverse": false,
        "listItem": false,
        "mask": null,
        "pull": null,
        "pulse": false,
        "rotation": null,
        "size": null,
        "spin": true,
        "symbol": false,
        "title": "",
        "transform": null,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
